### DISCUSSION TITLE
Import images using a dynamically interpolated name

### DISCUSSION THREAD

**Discussion by tomocrafter** [ElMassimo/vite_ruby#65] [Q&A] [answered]
I want to use images on component dynamically (which is filename of image file is not sure until the component mounted.)
but
```import(`@/assets/images/${name}.jpg`)```
occurs following error:
`TypeError: Failed to resolve module specifier '@/assets/images/filename.jpg'`
I want to locate the assets file within `app/frontend`, How can I import the assets file from it?

PS. If I import it by ```import(`../../assets/images/${name}.jpg`)``` then it will be successfully imported but images are not shown as expected(seems like it only fetching `export default <ASSET PATH>` by `<URL>/filename.jpg?import`).

Thank you! (I'm sorry if my English is wrong, Please feel free to ask me if any details needed.)

**Comment by nyrf** [comment]
I think you can submit a demo repo. you can look at https://vitejs.dev/guide/assets.html#explicit-url-imports or try `import.meta.globEager`

**Comment by ElMassimo** [comment, marked as answer]
As @nyrf suggested, the way to achieve that in Vite is by using `import.meta.glob` or `import.meta.globEager`.

```js
// app/frontend/utils/images.js

const images = import.meta.globEager('../assets/images/**/*')

export function getImageUrl (name) {
  const importModule = images[`../assets/images/${name}.jpg`]
  if (!importModule) throw new Error(`No image available for ${name}`)
  return importModule.default
}
```

Please see [this related discussion](https://github.com/ElMassimo/vite_ruby/discussions/26#discussioncomment-436073) where I describe a similar usage in more detail.
